pmacApp/src/pmacCSAxis.cpp

```diff
@@ -395,7 +395,7 @@ asynStatus pmacCSAxis::getAxisStatus(pmacCommandStore *sPtr) {
   previous_position_ = position;
   previous_direction_ = direction;
 
-  moving_ = !cStatus.done_ || deferredMove_;
+  moving_ = !cStatus.done_ || deferredMove_ || updateMove_;
```

Comment:
Move the position update to after of the moving, and check if the `position` is different from the `previous_position_`

```
  moving_ = !cStatus.done_ || deferredMove_ || (position != previous_position_);

  // Store position to calculate direction for next poll.
  previous_position_ = position;
  previous_direction_ = direction;
```